rlberry/agents/torch/dqn/dqn.py

```diff
@@ -1,568 +1,449 @@
-import torch
-from gym import spaces
 import logging
-import numpy as np
-from pathlib import Path
-import pickle
 
+import numpy as np
+import torch
+from gym import spaces
+from rlberry import types
 from rlberry.agents import AgentWithSimplePolicy
-from rlberry.agents.utils.memories import (
-    Transition,
-    PrioritizedReplayMemory,
-    TransitionReplayMemory,
-)
-from rlberry.exploration_tools.discrete_counter import DiscreteCounter
-from rlberry.exploration_tools.online_discretization_counter import (
-    OnlineDiscretizationCounter,
-)
-from rlberry.wrappers.uncertainty_estimator_wrapper import UncertaintyEstimatorWrapper
-from rlberry.agents.torch.dqn.exploration import exploration_factory
 from rlberry.agents.torch.utils.training import (
     loss_function_factory,
     model_factory,
-    size_model_config,
-    trainable_parameters,
     optimizer_factory,
+    size_model_config,
 )
-from rlberry.seeding import Seeder
-from rlberry.utils.factory import load
+from rlberry.agents.torch.dqn.dqn_utils import polynomial_schedule, lambda_returns
+from rlberry.agents.utils import replay
 from rlberry.utils.torch import choose_device
+from rlberry.utils.factory import load
+from typing import Callable, Optional, Union
+
 
 logger = logging.getLogger(__name__)
 
 
-def default_qvalue_net_fn(env):
+def default_q_net_fn(env, **kwargs):
     """
     Returns a default Q value network.
     """
-    model_config = {"type": "DuelingNetwork"}
+    del kwargs
+    model_config = {
+        "type": "MultiLayerPerceptron",
+        "layer_sizes": (64, 64),
+        "reshape": False,
+    }
     model_config = size_model_config(env, **model_config)
     return model_factory(**model_config)
 
 
 class DQNAgent(AgentWithSimplePolicy):
-    """
-    Deep Q Learning Agent.
+    """DQN Agent based on PyTorch.
+
+    Notes
+    -----
+    Uses Q(lambda) for computing targets by default. To recover
+    the standard DQN, set :code:`lambda_ = 0.0` and :code:`chunk_size = 1`.
 
     Parameters
     ----------
-    env: gym.Env
-        Environment
-    horizon : int
-        Maximum lenght of an episode.
-    gamma : double
-        Discount factor
-    qvalue_net_fn : function(env, **kwargs)
-        Function that returns an instance of a network representing
-        the Q function.
-        If none, a default network is used.
-    qvalue_net_kwargs:
-        kwargs for qvalue_net_fn
-    loss_function : str
-        Type of loss function. Possibilities: 'l2', 'l1', 'smooth_l1'
-    batch_size : int
-        Batch size
-    device : str
-        Device used by pytorch.
-    target_update : int
-        Number of steps to wait before updating the target network.
-    double : bool
-        If true, use double Q-learning.
-    learning_rate : double
+    env: :class:`rlberry.types.Env`
```

Comment:
I think you missed a tilde :class:`~rlberry.types.Env` in order to make the link.